FeedAPIChallenge/Feed API/FeedMapper.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  FeedMapper.swift
+//  FeedAPIChallenge
+//
+//  Created by Ahmed Atef Ali Ahmed on 02.07.21.
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+
+public protocol FeedImageMapper {
+	func map(_ data: Data, response: HTTPURLResponse) -> FeedLoader.Result
+}
+
+public final class RemoteFeedImageMapper: FeedImageMapper {
+	public static let `default` = RemoteFeedImageMapper()
+
+	private static let OK_STATUS_CODE = 200
+
+	public func map(_ data: Data, response: HTTPURLResponse) -> FeedLoader.Result {
+		guard response.statusCode == Self.OK_STATUS_CODE,
+		      let root = try? JSONDecoder().decode(Root.self, from: data)
+		else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+		return .success(root.images)
+	}
+}
```

Comment:
`RemoteFeedImageMapper` is an internal implementation detail of `RemoteFeedLoader`, thus it should not be accessible from outside of the module, please make it `internal`

Following from my other comments, the `map` function could be made `static` so there would be no need to create an instance of `RemoteFeedImageMapper` to call it.

```suggestion
final class RemoteFeedImageMapper: FeedImageMapper {
	private static let OK_STATUS_CODE = 200

	static func map(_ data: Data, response: HTTPURLResponse) -> FeedLoader.Result {
		guard response.statusCode == OK_STATUS_CODE,
		      let root = try? JSONDecoder().decode(Root.self, from: data)
		else {
			return .failure(RemoteFeedLoader.Error.invalidData)
		}
		return .success(root.images)
	}
}
```